### ISSUE TITLE
Why not choose e2e_res as our tune variable

### ISSUE THREAD

**Issue by cyk2018** [Dao-AILab/flash-attention#2237]
e2e_res and e2e_freq is very importent in FA4 as it decide the ratio between SFU and ALU. however e2e_res is not transfer through the function call, so now we choose to tune the e2e_freq as 16 and 32 to change the ratio. however the frag_tile is const 32. so e2e_freq keep 16 is better as we do 2 compute in one iterate. So I think if it is better to keep the e2e_freq 16 and tune e2e_res?

**Comment by tridao** [comment]
There are many hyperparameters (tile size, ordering of instructions, granularity of tmem loading/storing, granularity of smem loading / storing, overlapping of tmem, overlapping of smem), exposing all of them will make the interface too unwieldy. Generally we expose tile size (easy to understand) as a tunable parameter.

**Comment by cyk2018** [comment]
Thanks for your reply! So I know the reasons.By the way, if the tile size changed according to input? suchas prefill and decode?

**Comment by tridao** [comment]
Tile size can change depending on bf16 or fp8, causal or noncausal, H100 vs B200, etc.

**Comment by cyk2018** [comment]
Yes! as we are change the causal or not. my question is the frag_tile [here](https://github.com/Dao-AILab/flash-attention/blob/f1284cff5d2b2ad4160ceefaf096a800502d16fd/flash_attn/cute/softmax.py#L241C9-L241C22) is constant 32, so If I should think this frag_tile is not the "tile size". I think this cannot changed whether causal or not?

**Comment by tridao** [comment]
Nope that's not the tile size i'm referring to. On Hopper we use tile size 128 x 192 for noncausal and tile size 128 x 128 for causal.
https://github.com/Dao-AILab/flash-attention/blob/f1284cff5d2b2ad4160ceefaf096a800502d16fd/flash_attn/cute/interface.py#L276

**Comment by cyk2018** [comment]
Thanks for your reply, now I know how to found the block size. It seems not changed on blackwell.
